src/binary/writer.rs

```diff
@@ -636,8 +736,33 @@ impl<W: Write> BinarySystemWriter<W> {
         Ok(())
     }
 
+    fn set_field_name<A: AsRawSymbolTokenRef>(&mut self, name: A) {
+        match name.as_raw_symbol_token_ref() {
+            RawSymbolTokenRef::SymbolId(sid) => self.set_field_id(sid),
+            RawSymbolTokenRef::Text(text) => panic!(
+                "The RawBinaryWriter can only accept Symbol ID field names, not text ('{}').",
+                text
+            ),
+        }
+    }
+
+    fn parent_type(&self) -> Option<IonType> {
```

Comment:
When writing tooling, I've found lots of places where it would be nice to know the type of the parent container without having to track it myself. Some of our implementations have an `isInStruct` method (sometimes public, sometimes not); this is a more useful version of that that can also distinguish between `List` and `SExpression`.